optuna/integration/xgboost.py

```diff
@@ -43,6 +43,9 @@ class XGBoostPruningCallback(object):
             ``validation-merror``. Please refer to ``eval_metric`` in
             `XGBoost reference <https://xgboost.readthedocs.io/en/latest/parameter.html>`_
             for further details.
+            Note that when using the sklearn API callback, the index number of `eval_set`
```

Comment:
I have four comments on this line.

1. How about inserting this sentence before the previous line (e.g., `validation-merror`. When using the Scikit-Learn API, the index number of ...)? IMO, it will be easier to read if the explanation of ``observation_key`` is not separated. 
2. could you replace "sklearn API callback" with "Scikit-Learn API" since the [xgboost reference](https://xgboost.readthedocs.io/en/latest/python/python_api.html#module-xgboost.sklearn) calls it "Scikit-Learn" API?
3. How about adding a link to the reference? 
4. `eval_set` needs one more pair of backticks.

![image](https://user-images.githubusercontent.com/3255979/82550733-fa4eef00-9b99-11ea-8a9f-9df569d26d67.png)

Reply:
Thanks for your review. Totally agree with your suggestion of (1), (2), (4).
As for (3), it is not documented that `observation_key` is passed to the callback function in a format like `validation_0-merror`. So I don't think adding [references](https://xgboost.readthedocs.io/en/latest/python/python_api.html#module-xgboost.sklearn) is helpful. How do you think about it?

Reply:
Thank you for your update. 

> As for (3), it is not documented that `observation_key` is passed to the callback function in a format like `validation_0-merror`. 

That makes sense. I agree to your opinion.